src/jdk.accessibility/windows/classes/com/sun/java/accessibility/internal/AccessBridge.java

```diff
@@ -1735,6 +1757,20 @@ public AccessibleValue call() throws Exception {
         }, ac);
     }
 
+    /**
+     * return the screen coordinates transform from an AccessibleContext
+     */
+    private AffineTransform getTransformFromContext(final AccessibleContext ac) {
+        AccessibleComponent acomp = ac.getAccessibleComponent();
+        if (acomp != null) {
+            FontMetrics fm = acomp.getFontMetrics(acomp.getFont());
+            if (fm != null) {
+                return fm.getFontRenderContext().getTransform();
```

Comment:
I am not sure do we really update the FRC for the component when the DPI is changed?

Reply:
I've checked it - it does not change, which is unexpected... Doesn't it seem like a bug?

Actually, the problem is that AccessibleComponent does not provide the component's GC, which is a missing. Ideally would be to it to the API and then use - because it already contains actual scales for the component. It's a pity to do all that calculations just because we lack API calls...

Is it worth trying to fix the font metrics update instead and use it until the new API is ready and available?

Reply:
> I've checked it - it does not change, which is unexpected... Doesn't it seem like a bug?

Yes, I think this is a bug.

> Actually, the problem is that AccessibleComponent does not provide the component's GC, which is a missing. Ideally would be to add it to the API and then use - because it already contains actual scales for the component. It's a pity to do all that calculations just because we lack API calls...

But do we actually need the GC where the component is located, or do we need a GC where the a11y frame should be placed? I am not sure that both are always the same. Note that you can find the screen where the "virtual coordinates are located" and then convert them to the device space.

Reply:
> But do we actually need the GC where the component is located, or do we need a GC where the a11y frame should be placed? I am not sure that both are always the same. Note that you can find the screen where the "virtual coordinates are located" and then convert them to the device space.

Well, in `AccessBridge.getAccessibleContextAt_1(int x, int y, AccessibleContext parent)` we expect [x, y] within the parent bounds and so we can assume [x, y] has the same scaling. However, `AccessBridge.getAccessibleContextAt_2` works with generic [x, y] and there we have to perform generic transformation, indeed...

Ok, if we can't avoid it, then I'm ok with the latest version (taking into account my inline comments). Thanks!